File: app/src/main/java/com/plweegie/android/squash/data/RoomModule.java
package com.plweegie.android.squash.data;


import android.app.Application;
import androidx.room.Room;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class RoomModule {

    String mDatabaseName;

    public RoomModule(String databaseName) { this.mDatabaseName = databaseName;}

    @Provides
    @Singleton
    RepoDatabase provideDatabase(Application application) {
        return Room.databaseBuilder(application, RepoDatabase.class, this.mDatabaseName).build();
    }
}

File: app/src/main/java/com/plweegie/android/squash/data/RoomModule.kt
package com.plweegie.android.squash.data


import android.app.Application
import androidx.room.Room
import dagger.Module
import dagger.Provides
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors
import javax.inject.Singleton

@Module
class RoomModule(private val databaseName: String) {

    @Provides
    @Singleton
    fun provideDatabase(application: Application): RepoDatabase =
        Room.databaseBuilder(application, RepoDatabase::class.java, databaseName).build()

    @Provides
    @Singleton
    fun provideDiskIOExecutor(): ExecutorService = Executors.newSingleThreadExecutor()
}
